data-service/src/main/java/uk/gov/gchq/palisade/service/data/model/AuthorisedData.java

```diff
@@ -0,0 +1,179 @@
+/*
+ * Copyright 2018-2021 Crown Copyright
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package uk.gov.gchq.palisade.service.data.model;
+
+import uk.gov.gchq.palisade.Context;
+import uk.gov.gchq.palisade.Generated;
+import uk.gov.gchq.palisade.User;
+import uk.gov.gchq.palisade.resource.LeafResource;
+import uk.gov.gchq.palisade.rule.Rules;
+
+import java.util.Objects;
+import java.util.Optional;
+import java.util.StringJoiner;
+
+/**
+ * The AuthorisedData represents reference for the filtered resource that the user is authorised to retrieve.
+ * This will be used to provide the link to the resource that will be forwarded to the user to be streamed.
+ */
+public final class AuthorisedData {
```

Comment:
Following the pattern you've used elsewhere, this should be `AuthorisedDataRequest`
```suggestion
public final class AuthorisedDataRequest {
```

Reply:
The name of this class has been around the block a few times.  I'll give is go and see if that is better.